web/src/tippyjs.ts

```diff
@@ -84,9 +85,22 @@ export const topic_visibility_policy_tooltip_props = {
     onShow(instance: tippy.Instance) {
         const $elem = $(instance.reference);
         const current_visibility_policy_str = $elem.attr("data-tippy-content");
+        const should_render_privacy_icon = $elem
+            .attr("aria-label")
+            ?.startsWith("Notifications are based on your configuration for");
+        const privacy_policy = stream_data.get_stream_privacy_policy(
+            Number($elem.parent().attr("data-stream-id")),
+        );
+        const stream_name = $elem.parent(".change_visibility_policy").attr("data-stream-name");
         instance.setContent(
             ui_util.parse_html(
-                render_change_visibility_policy_button_tooltip({current_visibility_policy_str}),
+                render_change_visibility_policy_button_tooltip({
+                    current_visibility_policy_str,
+                    should_render_privacy_icon,
+                    stream_name,
+                    invite_only: privacy_policy.startsWith("invite-only"),
+                    is_web_public: privacy_policy.startsWith("web-public"),
```

Comment:
Why are these starts-with checks?

I think it'd be much safer to (1) do exact checks and (2) change `get_stream_privacy_policy` to declare a type with a fixed set of literal values, so the type-checker can help validate this code.

But I also think it'd be more normal to just fetch the stream object using `stream_data` and then access its `.invite_only` property, so probably `stream_data.get_stream_privacy_policy` shouldn't be used anyway.

Reply:
Redid the whole logic and now passing the stream obj.